### PR TITLE
Add Quay Workshop

### PR DESCRIPTION
**Pull request by mbach04**
##### SUMMARY

Adds a Quay Workshop for 1 user to RHPDS catalog

##### ISSUE TYPE
- Feature Pull Request
- New role Pull Request

##### COMPONENT NAME
ocp4_workload_quay_workshop

### CODE REVIEW

File: ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/tasks/workload.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  k8s:
+    state: present
+    definition:  "{{ lookup('template', item ) | from_yaml }}"
+  loop:
+    - ./templates/quay-ns.yml
+    - ./templates/homeroom-ns.yml
+    - ./templates/rolebinding.yml
+    - ./templates/operatorgroup.yml
+    - ./templates/subscription.yml
+    - ./templates/secret.yml
```
Comment (jkupferer) [inline ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/tasks/workload.yml:24]: No need to include `./templates/`, the templates directory is already implied for the lookup path when using `lookup('template', ...)`

File: ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/tasks/workload.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  command: oc apply -f https://raw.githubusercontent.com/afouladi7/quay-workshop/master/templates/hosted-workshop-production.json -n homeroom
+
+- name: Deploy the homeroom spawner
+  command: oc new-app hosted-workshop-production -n homeroom \
+          -p SPAWNER_NAMESPACE=homeroom \
+          -p CLUSTER_SUBDOMAIN={{ CLUSTER_SUBDOMAIN }} \
+          -p WORKSHOP_NAME=quay-workshop \
+          -p CONSOLE_IMAGE=quay.io/openshift/origin-console:4.6 \
+          -p WORKSHOP_IMAGE=quay.io/redhatgov/quay-workshop:latest \
+          -p CUSTOM_TAB_1=Webhooks=https://webhook.site
```
Comment (jkupferer) [inline ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/tasks/workload.yml:36]: Consider using a YAML block quote, `>-`, here:

```
- name: Deploy the homeroom spawner
  command: >-
    oc new-app hosted-workshop-production -n homeroom
    -p SPAWNER_NAMESPACE=homeroom
    -p CLUSTER_SUBDOMAIN={{ CLUSTER_SUBDOMAIN }}
    -p WORKSHOP_NAME=quay-workshop
    -p CONSOLE_IMAGE=quay.io/openshift/origin-console:4.6
    -p WORKSHOP_IMAGE=quay.io/redhatgov/quay-workshop:latest
    -p CUSTOM_TAB_1=Webhooks=https://webhook.site
```

File: ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/templates/homeroom-template.json [hunk 3]
```diff
@@ -0,0 +1,600 @@
+{
```
Comment (jkupferer) [inline ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/templates/homeroom-template.json:1]: Can this be YAML for consistency?

File: ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/templates/secret.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: redhat-pull-secret
+  namespace: quay
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: ewogICJhdXRocyI6IHsKICAgICJxdWF5LmlvIjogewogICAgICAiYXV0aCI6ICJjbVZrYUdGMEszRjFZWGs2VHpneFYxTklVbE5LVWpFMFZVRmFRa3MxTkVkUlNFcFRNRkF4VmpSRFRGZEJTbFl4V0RKRE5GTkVOMHRQTlRsRFVUbE9NMUpGTVRJMk1USllWVEZJVWc9PSIsCiAgICAgICJlbWFpbCI6ICIiCiAgICB9CiAgfQp9
```
Comment (jkupferer) [inline ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/templates/secret.yml:8]: This looks like a quay.io auth secret to be published in a public repository?

**Review by jkupferer — CHANGES_REQUESTED** [review]

File: ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/templates/secret.yml (hunk 4 above)
Comment (mbach04) [reply ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/templates/secret.yml:8]: Yes, that same auth secret is already publically published in a number of places.

File: ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/templates/homeroom-template.json (hunk 3 above)
Comment (mbach04) [reply ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/templates/homeroom-template.json:1]: yes

File: ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/tasks/remove_workload.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+---
+# Implement your workload removal tasks here
+# ------------------------------------------
+
+
+
+
+
```
Comment (jkupferer) [inline ansible/roles_ocp_workloads/ocp4_workload_quay_workshop/tasks/remove_workload.yml:8]: I believe yamllint will complain that there are too many blank lines here. To test, run `yamllint` with the current directory as `roles_ocp_workloads`.

**Review by jkupferer — CHANGES_REQUESTED** [review]

**Review by jkupferer — APPROVED** [review]

**Pull request merged**